exchanges/gateio/gateio_test.go

```diff
@@ -74,838 +73,2905 @@ func TestStart(t *testing.T) {
+	if err != nil {
+		t.Error(err)
+	}
+	_, err = g.GetRecentTrades(context.Background(), currencyPair, asset.Options)
+	if err != nil {
+		t.Error(err)
+	}
+}
+
+func TestSubmitOrder(t *testing.T) {
+	if !areTestAPIKeysSet() || !canManipulateRealOrders {
+		t.Skip(credInformationNotProvidedOrManipulatingRealOrdersNotAllowed)
+	}
+	enabledPairs, err := g.GetEnabledPairs(asset.CrossMargin)
+	if err != nil {
+		t.Fatal(err)
+	}
+	var orderSubmission = &order.Submit{
+		Exchange:  g.Name,
+		Pair:      enabledPairs[0],
+		Side:      order.Buy,
+		Type:      order.Limit,
+		Price:     1,
+		Amount:    1,
+		AssetType: asset.CrossMargin,
+	}
+	_, err = g.SubmitOrder(context.Background(), orderSubmission)
+	if err != nil {
+		t.Errorf("Order failed to be placed: %v", err)
+	}
+}
+
+func TestCancelExchangeOrder(t *testing.T) {
+	if !areTestAPIKeysSet() || !canManipulateRealOrders {
+		t.Skip(credInformationNotProvidedOrManipulatingRealOrdersNotAllowed)
+	}
+	var orderCancellation = &order.Cancel{
+		OrderID:       "1",
+		WalletAddress: core.BitcoinDonationAddress,
+		AccountID:     "1",
+		Pair:          currency.NewPair(currency.LTC, currency.BTC),
+		AssetType:     asset.Spot,
+	}
+	err := g.CancelOrder(context.Background(), orderCancellation)
+	if err != nil && !strings.Contains(err.Error(), "ORDER_NOT_FOUND") {
+		t.Errorf("%s CancelOrder error: %v", g.Name, err)
+	}
+}
+
+func TestCancelBatchOrders(t *testing.T) {
+	if !areTestAPIKeysSet() || !canManipulateRealOrders {
+		t.Skip(credInformationNotProvidedOrManipulatingRealOrdersNotAllowed)
+	}
+	enabledPairs, err := g.GetEnabledPairs(asset.Spot)
+	if err != nil {
+		t.Fatal(err)
+	}
+	_, err = g.CancelBatchOrders(context.Background(), []order.Cancel{
+		{
+			OrderID:       "1",
+			WalletAddress: core.BitcoinDonationAddress,
+			AccountID:     "1",
+			Pair:          enabledPairs[0],
+			AssetType:     asset.Spot,
+		}, {
+			OrderID:       "2",
+			WalletAddress: core.BitcoinDonationAddress,
+			AccountID:     "1",
+			Pair:          enabledPairs[1],
+			AssetType:     asset.Spot,
+		}})
+	if err != nil && !strings.Contains(err.Error(), "ORDER_NOT_FOUND") {
+		t.Errorf("%s CancelOrder error: %v", g.Name, err)
+	}
+}
+
+func TestGetDepositAddress(t *testing.T) {
+	if !areTestAPIKeysSet() {
+		t.Skip(credInformationNotProvided)
+	}
+	_, err := g.GetDepositAddress(context.Background(), currency.USDT, "", "TRX")
+	if err != nil {
+		t.Error("Test Fail - GetDepositAddress error", err)
+	}
+}
+
+func TestGetActiveOrders(t *testing.T) {
+	if !areTestAPIKeysSet() {
+		t.Skip(credInformationNotProvided)
+	}
+	enabledPairs, err := g.GetEnabledPairs(asset.Spot)
+	if err != nil {
+		t.Fatal(err)
+	}
+	var getOrdersRequest = order.GetOrdersRequest{
+		Pairs:     enabledPairs[:2],
+		Type:      order.AnyType,
+		Side:      order.AnySide,
+		AssetType: asset.Spot,
+	}
+	_, err = g.GetActiveOrders(context.Background(), &getOrdersRequest)
```

Comment:
Fee currency field is not being set to fee asset from last [pending nit](https://github.com/thrasher-corp/gocryptotrader/pull/1058#discussion_r1051690336)

raw response as reference:

```json
[{"currency_pair":"BTC_USDT","total":1,"orders":[{"id":"256380915255","text":"t-1206300907764","create_time":"1672874846","update_time":"1672874846","create_time_ms":1672874846805,"update_time_ms":1672874846805,"status":"open","currency_pair":"BTC_USDT","type":"limit","account":"spot","side":"buy","amount":"1","price":"1","time_in_force":"gtc","iceberg":"0","left":"1","fill_price":"0","filled_total":"0","fee":"0","fee_currency":"BTC","point_fee":"0","gt_fee":"0","gt_maker_fee":"0","gt_taker_fee":"0","gt_discount":false,"rebated_fee":"0","rebated_fee_currency":"USDT"}]},{"currency_pair":"MNW_USDT","total":2,"orders":[{"id":"268056881539","text":"t-9818787172","create_time":"1674518779","update_time":"1674518779","create_time_ms":1674518779653,"update_time_ms":1674518779653,"status":"open","currency_pair":"MNW_USDT","type":"limit","account":"spot","side":"sell","amount":"1","price":"100","time_in_force":"gtc","iceberg":"0","left":"1","fill_price":"0","filled_total":"0","fee":"0","fee_curren
cy":"USDT","point_fee":"0","gt_fee":"0","gt_maker_fee":"0","gt_taker_fee":"0","gt_discount":false,"rebated_fee":"0","rebated_fee_currency":"MNW"},{"id":"259259668238","text":"t-3582595394","create_time":"1673304746","update_time":"1673304746","create_time_ms":1673304746959,"update_time_ms":1673304746959,"status":"open","currency_pair":"MNW_USDT","type":"limit","account":"spot","side":"sell","amount":"1","price":"30","time_in_force":"gtc","iceberg":"0","left":"1","fill_price":"0","filled_total":"0","fee":"0","fee_currency":"USDT","point_fee":"0","gt_fee":"0","gt_maker_fee":"0","gt_taker_fee":"0","gt_discount":false,"rebated_fee":"0","rebated_fee_currency":"MNW"}]}]
```